docs/languages/csharp/event-pattern.md

```diff
@@ -1,9 +1,344 @@
+
+The second argument has typically been a type that is derived from
+`System.EventArgs`. (You'll see in the 
+[next section](modern-events.md) that this convention is no longer
+enforced.) If your event type does not need any additional
+arguments, you will still provide both arguments.
+There is a special value, `EventArgs.Empty` that you should use to
+denote that your event does not contain any additional information.
+
+Let's build a class that lists files in a directory, or any
+subdirectory that follow a pattern. This component raises an event
+for each file found that matches the pattern.
+
+Using an event model provides some design advantages. You can create
+multiple event listeners that perform different actions when a sought
+file is found. Combining the different listeners can create more
+robust algorithms.
+
+Here is the initial event argument declaration for finding a sought
+file: 
+
+```cs
+public class FileFoundArgs : EventArgs
+{
+    public string FoundFile { get; }
+
+    public FileFoundArgs(string fileName)
+    {
+        FoundFile = fileName;
+    }
+}
+```
+
+Even though this type looks like a small, data only type, you should
+follow the convention and make it a reference (`class`) type. That
+means the argument object will be passed by reference, and any
+updates to the data will be viewed by all subscribers. The first
+version is an immutable object. You should prefer to make the
+properties in your event argument type immutable. That way, one
+subscriber cannot change the values before another subscriber sees
+them. (There are exceptions to this, as you'll see below.)  
+
+Next, we need to create the event declaration in the FileSearcher
+class. Leveraging the `EventHandler<T>` type means that you don't
+need to create yet another type definition. You simply use a generic
+specialization.
+
+Let's fill out the FileSearcher class to search for files that match
+a pattern, and raise the correct event when a match is discovered.
+
+```cs
+public class FileSearcher
+{
+    public event EventHandler<FileFoundArgs> OnFoundFile;
+
+    public void Search(string directory, string searchPattern)
+    {
+        foreach (var file in Directory.EnumerateFiles(directory, searchPattern))
+        {
+            OnFoundFile?.Invoke(this, new FileFoundArgs(file));
+        }
+    }
+}
+```
+
+# Definining and Raising Field Like Events
+
+The simplest way to add an event to your class is to declare that
+event as a public field, as in the above example:
+
+```cs
+public event EventHandler<FileFoundArgs> OnFoundFile;
+```
+
+This looks like it's declaring a public field, which would appear to
+be bad Object Oriented practice. You want to protect data access
+through properties, or methods. While this make look like a bad
+practice, the code generated by the compiler does create wrappers so
+that the event objects can only be accessed in safe ways. The only
+operations available on a Field like event are add handler:
```

Comment:
field-like (lowercase and hyphenated)
set off the operation "add handler" in some way, maybe in double quotes to make it clear that it's an operation